src/embind/emval.js

```diff
@@ -77,23 +80,30 @@ var LibraryEmVal = {
         case true: return 3;
         case false: return 4;
         default:{
-          return emval_handles.allocate({refcount: 1, value: value});
+          const handle = emval_freelist.pop() || emval_handles.length / 2;
+          emval_handles[handle * 2] = value;
+          emval_handles[handle * 2 + 1] = 1;
+          return handle;
         }
       }
     }
   },
 
-  _emval_incref__deps: ['$emval_handles'],
+  _emval_incref__deps: ['$emval_handles', '$emval_handles_reserved'],
   _emval_incref: (handle) => {
-    if (handle > 4) {
-      emval_handles.get(handle).refcount += 1;
+    if (handle >= emval_handles_reserved) {
```

Comment:
Should we just remove this line and initialize all the the reserved handles to refcount == 1? 

That will safe on code size and remove an extra branch here.. at the cost of redundantly storing/tracking the refcount of the reserved types (could could assert that the refcount of a reserved type never hits zero too).

Reply:
Was also on my mind, but was trying to minimize the changes. Seems better to minimize the code paths though.